### PR TITLE
convert glue.c to glue.d

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by WalterBright** [comment]
Blocked by https://github.com/dlang/dmd/pull/6139

**Comment by UplinkCoder** [comment]
Very nice. I cannot wait to have this in master!

File: src/backend/code.d [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+struct FuncParamRegs
+{
+    //this(tym_t tyf);
+    static FuncParamRegs create(tym_t tyf);
+
+    int alloc(type *t, tym_t ty, ubyte *reg1, ubyte *reg2);
+
+  private:
+    tym_t tyf;                  // type of function
```
Comment (UplinkCoder) [inline src/backend/code.d]: I wouldn't hurt to add another `/` so they show up in the docs

Comment (WalterBright) [inline src/backend/code.d]: When doing translations, I try hard to do as few changes as possible. I don't do any fixes, change comments, etc. I just want to get it to work - doing as few changes as possible makes it much, much easier to track down where a translation went wrong.

Fixes, improvements, converting to D idiomatic constructs, etc., all come after it is working.

File: src/glue.d [hunk 2]
```diff
@@ -242,93 +262,97 @@ void obj_start(char *srcfile)
...
 
-#if TARGET_WINDOS
-    // Produce Ms COFF files for 64 bit code, OMF for 32 bit code
-    assert(objbuf.size() == 0);
-    objmod = global.params.mscoff ? MsCoffObj::init(&objbuf, srcfile, NULL)
-                                  :       Obj::init(&objbuf, srcfile, NULL);
-#else
-    objmod = Obj::init(&objbuf, srcfile, NULL);
-#endif
+    version (Windows)
```
Comment (jacob-carlborg) [inline src/glue.d]: We have used `static if (TARGET_WINDOS)` if all other cases.

**Comment by UplinkCoder** [comment]
@jacob-carlborg do a pr :)

File: src/glue.d (hunk 2 above)
Comment (WalterBright) [inline src/glue.d]: Right, but the trouble here is the object file support for other targets is build-specific.

Comment (jacob-carlborg) [inline src/glue.d]: Hmm not sure I understand.

**Pull request merged**